### PR TITLE
Move Old HttpWebRequest to HttpClient Instance

### PR DESCRIPTION
**Pull request by taooceros**
This pull request changes the Get & Download method implementation in Flow.Launcher.Infrastructure.Http by moving the old HttpWebRequest API to new HttpClient.

This has fixed my issue of connection to Github while checking for new Update.

### CODE REVIEW

File: Flow.Launcher.Infrastructure/Http/Http.cs [hunk 1]
```diff
@@ -6,85 +6,112 @@
...
+                ProxyProperty.Port => (new Uri($"http://{Proxy.Server}:{Proxy.Port}"), WebProxy.Credentials),
+                ProxyProperty.UserName => (WebProxy.Address, new NetworkCredential(Proxy.UserName, Proxy.Password)),
+                ProxyProperty.Password => (WebProxy.Address, new NetworkCredential(Proxy.UserName, Proxy.Password)),
+                _ => throw new ArgumentOutOfRangeException()
+            };
+        }
+
+        public static async Task Download([NotNull] string url, [NotNull] string filePath)
+        {
+            using var response = await client.GetAsync(url);
```
Comment (jjw24) [inline Flow.Launcher.Infrastructure/Http/Http.cs:80]: client.GetAsync crashes the entire app for me when i try to install plugin. Cant see any reason why though, no exception is thrown

Comment (taooceros) [reply Flow.Launcher.Infrastructure/Http/Http.cs:80]: Let me take a check.

Comment (taooceros) [reply Flow.Launcher.Infrastructure/Http/Http.cs:80]: Oh that's because before the Http.Download is not async, so no await is added there.

**Comment by jjw24** [comment]
getting this error also when i check for update:

image

**Comment by taooceros** [comment]
> getting this error also when i check for update:
> 
> image

fix it, and the previous one. Forget to commit seperately, so both is on last commit.

**Comment by taooceros** [comment]
Hold on, update also call the method Http.Download so need to change that as well. Though, since we have InstallOrUpdate method, why not simply call that one instead of downloading newest release and install.

**Comment by taooceros** [comment]
image
emmm a question. Why do we need the mainthread to exercute this method?

**Comment by taooceros** [comment]
Done, please review. @jjw24 
and it seems that we don't need to invoke Install method into main thread......

File: Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources/Baidu.cs [hunk 2]
```diff
@@ -22,9 +23,9 @@ public override async Task<List<string>> Suggestions(string query)
             try
             {
                 const string api = "http://suggestion.baidu.com/su?json=1&wd=";
-                result = await Http.Get(api + Uri.EscapeUriString(query), "GB2312");
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources/Baidu.cs:25]: removal of "GB2312" is intentional?

**Comment by jjw24** [comment]
good work

**Comment by jjw24** [comment]
for reference on why use ConfigureAwait(false)

https://medium.com/bynder-tech/c-why-you-should-use-configureawait-false-in-your-library-code-d7837dce3d7f

File: Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources/Baidu.cs (hunk 2 above)
Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources/Baidu.cs:25]: Yes, I have checked, the ReadAsString doesn't need the encoding to create the result.

Comment (jjw24) [reply Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources/Baidu.cs:25]: do we need to keep  "GB2312"?

**Review by jjw24 — APPROVED** [review]

**Pull request merged**